File: python/1164.py
def perfeito (n):
    i=1
    cont=0
    while (i < n):
        if (n % i == 0):
            cont += i
        i+=1
    
    if cont == n:
        print(f"{n} eh perfeito")
    else:
        print(f"{n} nao eh perfeito")
    



if __name__:
    n = int(input())
    for i in range(0, n):
        x = int(input())
        perfeito(x)

File: js/1164.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8')
var lines = input.split('\n')
lines.map(x => parseInt(x))
n = lines.shift()

for(let i=0;i<n;i++){
    x = lines.shift()
    perfeito(x)
}

function perfeito (n){
    i=1, cont=0
    while (i < n) {
        if (n % i == 0) {
            cont += i;
        }
        i++;
    }
    if (cont == n) {
        console.log(n + " eh perfeito");
    } else {
        console.log(n + " nao eh perfeito");
    }
}